apps/condo/domains/user/schema/CheckUserExistenceService.js

```diff
@@ -0,0 +1,130 @@
+    },
+    WRONG_SENDER_FORMAT: {
+        query: 'checkUserExistence',
+        variable: ['data', 'sender'],
+        code: BAD_USER_INPUT,
+        type: WRONG_FORMAT,
+        message: 'Invalid format of "sender" field value',
+        correctExample: '{ dv: 1, fingerprint: \'example-fingerprint-alphanumeric-value\'}',
+    },
+}
+
+const CheckUserExistenceService = new GQLCustomSchema('CheckUserExistenceService', {
+    types: [
+        {
+            access: true,
+            type: 'input CheckUserExistenceInput { dv: Int!, sender: JSON!, confirmActionToken: String!, userType: UserTypeType! }',
+        },
+        {
+            access: true,
+            type: 'type CheckUserExistenceOutput { userIsExist: Boolean!, nameIsSet: Boolean!, emailIsSet: Boolean!, phoneIsSet: Boolean!, passwordIsSet: Boolean! }',
+        },
+    ],
+
+    queries: [
+        {
+            access: access.canCheckUserExistence,
+            schema: 'checkUserExistence(data: CheckUserExistenceInput!): CheckUserExistenceOutput',
+            doc: {
+                summary: 'Using an action token with a verified phone number,' +
+                    ' checks whether a user with the specified type (resident, staff, service) is registered.\n' +
+                    'As a result, information about the completion of some important fields (name, email, phone, password) is also returned.',
+                errors: ERRORS,
+            },
+            resolver: async (parent, args, context) => {
+                const { data } = args
+                const { confirmActionToken, userType } = data
+
+                checkDvAndSender(data, ERRORS.DV_VERSION_MISMATCH, ERRORS.WRONG_SENDER_FORMAT, context)
+
+                if (!confirmActionToken) throw new GQLError(ERRORS.TOKEN_NOT_FOUND, context)
+
+                const action = await ConfirmPhoneAction.getOne(context,
+                    {
+                        token: confirmActionToken,
+                        expiresAt_gte: new Date().toISOString(),
+                        completedAt: null,
+                        isPhoneVerified: true,
+                    },
+                )
+                if (!action) throw new GQLError(ERRORS.TOKEN_NOT_FOUND, context)
+
+                const phoneFromAction = normalizePhone(action.phone) || null
+                if (!phoneFromAction) throw new GQLError(ERRORS.INVALID_PHONE_NUMBER, context)
+
+                const users = await find('User', { type: userType, phone: action.phone })
+                if (users.length > 1) throw new GQLError(ERRORS.MULTIPLE_USERS_FOUND, context)
+
+                const user = users[0]
+                const result = {
+                    userIsExist: false,
+                    nameIsSet: false,
+                    emailIsSet: false,
+                    phoneIsSet: false,
+                    passwordIsSet: false,
+                }
+
+                if (!user || user.deletedAt) return result
```

Comment:
Correct me if I'm wrong, but deletedAt is used as "block mechanism". (User cannot relogin / reregister, if deletedAt and his phone  is set to non-null

Reply:
It's important for us to keep this behaviour, can we add test for this?

Reply:
> Correct me if I'm wrong, but deletedAt is used as "block mechanism". (User cannot relogin / reregister, if deletedAt and his phone is set to non-null

Yes, a user who was soft deleted" cannot sign in or register again

Reply:
> It's important for us to keep this behaviour, can we add test for this?

Added tests